android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java

```diff
@@ -0,0 +1,32 @@
+package com.liferay.mobile.screens.base;
+
+import android.util.Log;
+
+import com.liferay.mobile.screens.context.LiferayServerContext;
+
+/**
+ * @author Javier Gamarra
+ */
+public class FactoryProvider {
+
+	public static AbstractFactory getInstance() {
+		if (_abstractFactory == null) {
+			synchronized ("factory") {
```

Comment:
Yep, that's the equivalent of a static field in the class so just 1 thread could be entering, but it doesn't follow the conventions, changed to FactoryProvider.class.